src/field.jl

```diff
@@ -358,14 +364,41 @@ function tryparsenext(s::StringToken{T}, str, i, len, opts) where {T}
         y2 = iterate(str, i)
     end
 
-    return R(_substring(T, str, i0, i-1)), i
+    return R(_substring(T, str, i0, i-1, escapecount, opts.escapechar, opts.quotechar, opts.includequotes)), i
 end
 
-@inline function _substring(::Type{String}, str, i, j)
-    String(str[i:thisind(str, j)])
+@inline function _substring(::Type{String}, str, i, j, escapecount, escapechar, quotechar, includequotes)
+    if escapecount > 0
+        buf = IOBuffer(sizehint=j-i+1-escapecount)
+        cur_i = i
+        c = str[cur_i]
+        if includequotes && c==quotechar
+            print(buf, c)
+            cur_i = nextind(str, cur_i)
+        end
+        while cur_i <= j
+            c = str[cur_i]
+            if c == escapechar
+                next_i = nextind(str, cur_i)
+                if next_i <= j && str[next_i] == quotechar
+                    print(buf, str[next_i])
+                    cur_i = next_i
+                else
+                    print(buf, c)
+                end
+            else
+                print(buf, c)
+            end
+            cur_i = nextind(str, cur_i)
+        end
+        return String(take!(buf))
+    else
+        return unsafe_string(pointer(str, i), j-i+1)
+    end
 end
 
-@inline function _substring(::Type{T}, str, i, j) where {T<:SubString}
+@inline function _substring(::Type{T}, str, i, j, escapecount, escapechar, quotechar, includequotes) where {T<:SubString}
```

Comment:
why not just take opts as the last argument?

Reply:
Yep, done!